### ISSUE TITLE
Error Path does not exist with Metrics extension

### ISSUE THREAD

**Issue by mniemela** [MetricsGrimoire/CVSAnalY#85]
I tried Metrics extension with a simple test repository (test repository in issue 64 thread: https://github.com/MetricsGrimoire/CVSAnalY/blob/master/tests/input.tar.gz ) and got this error

Error obtaining bbb/something@c0d66f9. Command ['git', 'show', u'c0d66f92a95e31c77be08dc9d0f11a16715d1885:bbb/something'] returned 128 (fatal: Path 'bbb/something' does not exist in 'c0d66f92a95e31c77be08dc9d0f11a16715d1885'
)

Using Metrics with a large repository caused lots of similar errors.

**Comment by mniemela** [comment]
Looks like c0d66f92a95e31c77be08dc9d0f11a16715d1885 is the commit where file bbb/something was renamed to bbb/something.renamed.

**Comment by mniemela** [comment]
I added some test outputs to Metrics.py a few lines before the error is printed in code, and I got confusing results. The test ouputs were different during different runs, does the extension use several threads or something?

This output makes sense, as path is same in test outputs and the error:

path: bbb/something
self.path: bbb/something
Error obtaining bbb/something@c0d66f92a95e31c77be08dc9d0f11a16715d1885. Command ['git', 'show', u'c0d66f92a95e31c77be08dc9d0f11a16715d1885:bbb/something'] returned 128 (fatal: Path 'bbb/something' does not exist in 'c0d66f92a95e31c77be08dc9d0f11a16715d1885'
)

However, this doesn't make any sense:
path: bbb/something.renamed
self.path: bbb/something.renamed
Error obtaining bbb/something@c0d66f92a95e31c77be08dc9d0f11a16715d1885. Command ['git', 'show', u'c0d66f92a95e31c77be08dc9d0f11a16715d1885:bbb/something'] returned 128 (fatal: Path 'bbb/something' does not exist in 'c0d66f92a95e31c77be08dc9d0f11a16715d1885'
)

**Comment by sduenas** [comment]
I found the problem. This is related to some code that we removed some months ago. We didn't remember what that code was supposed to do, so we deleted it on 45b31b0e28923b89cf5601fe7cf9fc34128d9f51 (see the discussion here #53)

Metrics.py uses FileRevs.py to get the revisions ids (or commits) where each file was added, modified, etc. The problem is it also returns revisions where files were deleted. So, when Metrics extension receives a revision were the file was deleted, it tries to checkout that file instead of ignoring it.

To fix this issue I think I'm going to move the deleted code from FileRevs.py to Metrics.py. It's the best solution and doesn't break anything we did on FileRevs.
